eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/ExtendedBitcoinClient.scala

```diff
@@ -84,6 +89,41 @@ class ExtendedBitcoinClient(val rpcClient: BitcoinJsonRPCClient) {
       index = txs.indexOf(JString(txid.toHex))
     } yield (height.toInt, index)
 
+  def fundTransaction(tx: Transaction, options: FundTransactionOptions)(implicit ec: ExecutionContext): Future[FundTransactionResponse] = {
+    rpcClient.invoke("fundrawtransaction", tx.toString(), options).map(json => {
+      val JString(hex) = json \ "hex"
+      val JInt(changePos) = json \ "changepos"
+      val JDecimal(fee) = json \ "fee"
+      val fundedTx = Transaction.read(hex)
+      val amountIn = toSatoshi(fee) + fundedTx.txOut.map(_.amount).sum
```

Comment:
👀 

Can you make `amountIn` a method of `FundTransactionResponse`? I'd prefer we'd keep the response object a mirror of what bitcoin core returns.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1697/commits/b89cd280923c39e1f9207c2368e79c3ba3e9d21b